### DISCUSSION TITLE
macos / ssh-agent / 1password

### DISCUSSION THREAD

**Discussion by Robert-M-Muench** [iterate-ch/cyberduck#15571] [Q&A] [answered]
I have a 1P SSH key agent set up, but CD always asks me for a password. I don't know how to tell or find out if CD uses the SSH agent at all. How do I configure a connection to retrieve the private key from 1P?

**Comment by dkocher** [comment]
You will need to configure 1Password as an identity agent [^1] which is documented as interoperable with Cyberduck [^2].

[^1]: https://developer.1password.com/docs/ssh/agent/compatibility/#identity-agent
[^2]: https://developer.1password.com/docs/ssh/agent/compatibility/#cyberduck

**Comment by Robert-M-Muench** [comment]
I did that, but I'm still asked for a username/password. The agent doesn't seem to get queried.

**Comment by Robert-M-Muench** [comment]
When accessing the server via the CLI, the 1P agent is used and pops-up, and everything works.

**Comment by dkocher** [comment]
You can make sure to authenticate with the correct identity retrieved from the SSH agent by selecting the public key in the bookmark configuration _SSH Private Key_ or indicate it in the `IdentityFile` your OpenSSH ~/.ssh/config file [^1].

[^1]: https://docs.cyberduck.io/protocols/sftp/#public-key-authentication-using-ssh-agent

**Comment by Robert-M-Muench** [comment, marked as answer]
The problem was that the entered user name missed a character...🙄
